java-checks/src/test/files/checks/InnerStaticClassesCheck.java

```diff
@@ -0,0 +1,93 @@
+
+class Fruit {
+  private int value;
+
+  public class Seed {  // Noncompliant {{Make this a "static" inner class.}}
+    int germinationDays = 0;
+    public Seed(int germinationDays) {
+      this.germinationDays = germinationDays;
+    }
+    public int getGerminationDays() {
+      return germinationDays;
+    }
+    public class SeedI {  // Compliant, inner classes cannot be static
+      int germinationDays = 0;
+      public SeedI(int germinationDays) {
+        this.germinationDays = germinationDays;
+      }
+      public int getGerminationDays() {
+        return germinationDays;
+      }
+    }
+  }
+
+  public static class StaticSeed {
+    int germinationDays = 0;
+    public StaticSeed(int germinationDays) {
+      this.germinationDays = germinationDays;
+    }
+    public int getGerminationDays() {
+      return germinationDays;
+    }
+    public static class StaticSeedI {
+      int germinationDays = 0;
+      public StaticSeedI(int germinationDays) {
+        this.germinationDays = germinationDays;
+      }
+      public int getGerminationDays() {
+        return germinationDays;
+      }
+    }
+    public class StaticSeedI2 { // Noncompliant {{Make this a "static" inner class.}}
+
+    }
+  }
+
+  public class Referencing {
+    public Referencing() {
+      value = 5;
+    }
+    public class ReferencingI {
+      public ReferencingI() {
+        value = 5;
+      }
+    }
+  }
+
+  public class Referencing2 {
+    public Referencing2() {
+      Fruit.this.value = 6;
+    }
+    public class Referencing2I {
+      public Referencing2I() {
+        Fruit.this.value = 6;
+      }
+    }
+  }
+
+  // ************** False positive
+  public class Refering3 { // Noncompliant
```

Comment:
This is a false positive, static cannot be added because Referencing2 needs access to the instance, I don't have a nice idea to detect that, any idea?